src/audio/copier/copier.c

```diff
@@ -987,15 +959,8 @@ static int copier_comp_trigger(struct comp_dev *dev, int cmd)
 
 	for (i = 0; i < cd->endpoint_num; i++) {
 		if (dev->ipc_config.type == SOF_COMP_HOST) {
-			ret = comp_set_state(cd->endpoint[i], cmd);
-			if (ret < 0)
-				break;
 
-			if (ret == COMP_STATUS_STATE_ALREADY_SET) {
-				ret = PPL_STATUS_PATH_STOP;
-				break;
-			}
-			ret = host_zephyr_trigger(cd->hd, dev, cmd, cd->endpoint[i]->state);
+			ret = host_zephyr_trigger(cd->hd, dev, cmd, dev->state);
```

Comment:
if dev is passed why do we need to pass dev->state now?

Reply:
ok, this will be changed in previous patch.